applications/StructuralMechanicsApplication/custom_constitutive/yield_surfaces/tresca_yield_surface.h

```diff
@@ -0,0 +1,322 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:         BSD License
+//                   license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Alejandro Cornejo & Lucia Barbu
+//
+
+#if !defined(KRATOS_TRESCA_YIELD_SURFACE_H_INCLUDED)
+#define KRATOS_TRESCA_YIELD_SURFACE_H_INCLUDED
+
+// System includes
+
+// Project includes
+#include "custom_constitutive/yield_surfaces/generic_yield_surface.h"
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+/**
+ * @class TrescaYieldSurface
+ * @ingroup StructuralMechanicsApplication
+ * @brief
+ * @details
+ * @tparam TPlasticPotentialType The plastic potential considered
+ * @tparam TVoigtSize The number of components on the Voigt notation
+ * @author Alejandro Cornejo & Lucia Barbu
+ */
+template <class TPlasticPotentialType>
+class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) TrescaYieldSurface
```

Comment:
I think the KRATOS_API is not needed for header only files

If it compiles it is ok for me